Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/nbclient/client.py", line 1319, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# %%writefile plot_derivs.py

import numpy as np
import matplotlib.pyplot as plt
from scipy.misc import derivative as spderiv

def forward(f, x, h):
    return (f(x+h)-f(x))/h

def central(f, x, h):
    return (f(x+h/2)-f(x-h/2))/h

def central_richardson(f, x, h):
    # YOUR CODE HERE
    raise NotImplementedError()

def compare_derivatives():    
    fig, ax = plt.subplots()
    h = np.power(10.0, np.arange(-18, 0, +1))
    x = np.pi/3
    # YOUR CODE HERE
    raise NotImplementedError()
    

compare_derivatives()


------------------

----- stderr -----
/tmp/ipykernel_2270/2607954013.py:5: DeprecationWarning: scipy.misc is deprecated and will be removed in 2.0.0
  from scipy.misc import derivative as spderiv
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mImportError[0m                               Traceback (most recent call last)
Cell [0;32mIn[1], line 5[0m
[1;32m      3[0m [38;5;28;01mimport[39;00m[38;5;250m [39m[38;5;21;01mnumpy[39;00m[38;5;250m [39m[38;5;28;01mas[39;00m[38;5;250m [39m[38;5;21;01mnp[39;00m
[1;32m      4[0m [38;5;28;01mimport[39;00m[38;5;250m [39m[38;5;21;01mmatplotlib[39;00m[38;5;21;01m.[39;00m[38;5;21;01mpyplot[39;00m[38;5;250m [39m[38;5;28;01mas[39;00m[38;5;250m [39m[38;5;21;01mplt[39;00m
[0;32m----> 5[0m [38;5;28;01mfrom[39;00m[38;5;250m [39m[38;5;21;01mscipy[39;00m[38;5;21;01m.[39;00m[38;5;21;01mmisc[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m derivative [38;5;28;01mas[39;00m spderiv
[1;32m      7[0m [38;5;28;01mdef[39;00m[38;5;250m [39m[38;5;21mforward[39m(f, x, h):
[1;32m      8[0m     [38;5;28;01mreturn[39;00m (f(x[38;5;241m+[39mh)[38;5;241m-[39mf(x))[38;5;241m/[39mh

[0;31mImportError[0m: cannot import name 'derivative' from 'scipy.misc' (/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/scipy/misc/__init__.py)

